### PR TITLE
Throw a RootViolationException if the root folder does not exist and it'...

### PR DESCRIPTION
**Pull request by andris-sevcenko**
...s not possible to create it in Local Adapter.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please indent with spaces.

File: src/Adapter/Local.php [hunk 1]
```diff
@@ -40,12 +41,18 @@ public function __construct($root)
      *
      * @param string $root root directory path
      *
-     * @return string real path to root
+	 * @throws RootViolationException
```
Comment (staabm) [inline src/Adapter/Local.php]: Indentation

**Comment by frankdejonge** [comment]
Does is_dir return false when a dir exists and it's writable? Because that's what this implies. If not (which I think is the case), the writable check should be moved to after the first if statement.

**Comment by andris-sevcenko** [comment]
Since is_dir would return false on non-existing paths, say, "/this/path/does/not/exist", the code would then try to create the non-existing path. If the path would not be writable, it would result in a warning from mkdir. Throwing an exception there, if the folder does not exist and the path is not writable, allows for a more graceful handling of it.

**Comment by frankdejonge** [comment]
The problem is that you're checking for the wrong folder. A is_writable on a non-existing folder, which is what you're checking for, always results into `false`. So mkdir can, in this setup, never be called.

**Comment by andris-sevcenko** [comment]
@frankdejonge Ah I see. You're right. The cleanest way to perform this check then, in my opinion, would be to suppress the errors when creating the folder and then checking if the folder exists afterwards. If not, then throw the exception so it can be handled. Does that sound OK?

**Comment by frankdejonge** [comment]
@andris-sevcenko iirc, the mkdir function doesn't emit an error. Throwing an exception what it fails seems like a good idea though! :+1:

**Comment by andris-sevcenko** [comment]
Suppressing errors suppresses warnings as well :)

**Comment by GrahamCampbell** [comment]
Indentation is still incorrect. Also, this should be squashed to one commit.

**Comment by andris-sevcenko** [comment]
Right. Will resubmit a clean pull request.

**Pull request closed without merging**